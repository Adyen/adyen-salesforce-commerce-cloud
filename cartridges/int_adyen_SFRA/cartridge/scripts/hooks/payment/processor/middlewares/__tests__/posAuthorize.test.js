"use strict";

/* eslint-disable global-require */
var posAuthorize;
var orderNumber;
var paymentInstrument;
var paymentProcessor;
var Logger;
beforeEach(function () {
  posAuthorize = require('../posAuthorize');
  orderNumber = 'mockedNum';
  paymentInstrument = {
    paymentTransaction: {}
  };
  paymentProcessor = 'mockedPaymentProcessor';
  Logger = require('dw/system/Logger');
  jest.clearAllMocks();
});
afterEach(function () {
  jest.resetModules();
});
describe('POS Authorize', function () {
  it('should return error if there is no terminal ID', function () {
    var getForm = require('server').forms.getForm;
    getForm.mockImplementation(function () {
      return {
        adyenPaymentFields: {
          terminalId: {
            value: null
          }
        }
      };
    });
    var authorizeResult = posAuthorize(orderNumber, paymentInstrument, paymentProcessor);
    expect(authorizeResult).toMatchSnapshot();
    expect(Logger.fatal.mock.calls).toMatchSnapshot();
  });
  it('should return error if createTerminalPayment fails', function () {
    var _require = require('*/cartridge/scripts/adyenTerminalApi'),
      createTerminalPayment = _require.createTerminalPayment;
    createTerminalPayment.mockImplementation(function () {
      return {
        error: true,
        response: 'mockedResponse'
      };
    });
    var authorizeResult = posAuthorize(orderNumber, paymentInstrument, paymentProcessor);
    expect(authorizeResult).toMatchSnapshot();
    expect(Logger.fatal.mock.calls).toMatchSnapshot();
  });
  it('should return success response when createTerminalPayment passes', function () {
    var authorizeResult = posAuthorize(orderNumber, paymentInstrument, paymentProcessor);
    expect(authorizeResult).toMatchSnapshot();
  });
});