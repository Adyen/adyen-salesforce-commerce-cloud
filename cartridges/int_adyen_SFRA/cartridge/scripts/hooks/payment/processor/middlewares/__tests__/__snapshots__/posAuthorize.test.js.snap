// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POS Authorize should return error if createTerminalPayment fails 1`] = `
{
  "authorized": false,
  "error": true,
  "fieldErrors": [],
  "serverErrors": [
    "mocked_error.payment.processor.not.supported",
  ],
}
`;

exports[`POS Authorize should return error if createTerminalPayment fails 2`] = `
[
  [
    "POS Authorise error, result: mockedResponse",
  ],
]
`;

exports[`POS Authorize should return error if there is no terminal ID 1`] = `
{
  "authorized": false,
  "error": true,
  "fieldErrors": [],
  "serverErrors": [
    "mocked_error.payment.processor.not.supported",
  ],
}
`;

exports[`POS Authorize should return error if there is no terminal ID 2`] = `
[
  [
    "No terminal selected",
  ],
]
`;

exports[`POS Authorize should return success response when createTerminalPayment passes 1`] = `
{
  "response": "mockedSuccessResponse",
}
`;
